google/cloud/internal/log_wrapper.h

```diff
@@ -57,6 +59,47 @@ struct IsFutureStatus : public std::false_type {};
 template <>
 struct IsFutureStatus<future<Status>> : public std::true_type {};
 
+Status LogResponse(Status response, absl::string_view prefix,
+                   absl::string_view args, TracingOptions const& options);
+
+template <typename T>
+StatusOr<T> LogResponse(StatusOr<T> response, absl::string_view prefix,
+                        absl::string_view args, TracingOptions const& options) {
+  if (!response) {
+    GCP_LOG(DEBUG) << prefix << args
+                   << " >> status=" << DebugString(response.status(), options);
+  } else {
+    GCP_LOG(DEBUG) << prefix << args
+                   << " >> response=" << DebugString(*response, options);
+  }
+  return response;
+}
+
+future<Status> LogResponse(future<Status> response, std::string prefix,
+                           std::string args, TracingOptions const& options);
+
+template <typename T>
+future<StatusOr<T>> LogResponse(future<StatusOr<T>> response,
+                                std::string prefix, std::string args,
+                                TracingOptions options) {
+  GCP_LOG(DEBUG) << prefix << " >> future_status="
+                 << DebugFutureStatus(
+                        response.wait_for(std::chrono::microseconds(0)));
+  return response.then([prefix = std::move(prefix), args = std::move(args),
+                        options = std::move(options)](auto f) {
+    return LogResponse(f.get(), prefix, args, options);
+  });
+}
+
+template <typename T>
+std::unique_ptr<T> LogResponse(std::unique_ptr<T> response,
+                               absl::string_view prefix, absl::string_view args,
+                               TracingOptions const& /*options*/) {
+  GCP_LOG(DEBUG) << prefix << args << " >> " << (response ? "not null" : "null")
+                 << " stream";
```

Comment:
This could forward to another helper to get more stuff out of the per-`T` instantiations.

Reply:
Done.  I also did the same for `std::future_status`.